smarts/env/wrappers/record.py

```diff
@@ -0,0 +1,101 @@
+# Copyright (C) 2022. Huawei Technologies Co., Ltd. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+import os
+import shutil
+from typing import Any, Dict, Tuple
+
+import gym
+import numpy as np
+
+from smarts.core.sensors import Observation
+from smarts.core.utils.file import smarts_log_dir
+from smarts.env.wrappers.utils.rendering import vis_sim_obs
+
+Action = Any
+Operation = Any
+
+default_log_dir = smarts_log_dir()
+
+
+class AgentCameraRGBRender(gym.Wrapper):
```

Comment:
It might be better if the render functionality in this wrapper is moved directly into the render method of the base environment `SMARTS/smarts/env/hiway_env.py` and enabled/disabled according to `agent_interfaces` attribute. This will be more in line with how `env` and `env.render` are created in `gym`, and will remove the need for a separate wrapper.